client/http/client.go

```diff
@@ -30,15 +31,23 @@ type HttpClientConfig struct {
 }
 
 func NewHttpClient(config HttpClientConfig) (*HttpClient, error) {
-	return &HttpClient{
+	httpClient := &HttpClient{
 		ApiKey:    config.ApiKey,
 		ApiSecret: config.ApiSecret,
 		client:    config.RestClient.SetHostURL(config.ApiEndpoint).SetHeader("User-Agent", config.UserAgent),
-	}, nil
+	}
+	req := httpClient.client.R().SetBasicAuth(httpClient.ApiKey, httpClient.ApiSecret)
+	response, err := req.SetQueryParams(map[string]string{"encoded": "true"}).Get("auth/token")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
can we extract it into a new method please? would make code more descriptive...

Reply:
[refactoring](https://github.com/env0/terraform-provider-env0/pull/495/commits/23c4177a5c02f1bae871715ecc1a8ddf0ce24809)